src/app/DTT/flag/application_detail_view_flag.js

```diff
@@ -0,0 +1,82 @@
+import { Box, Button, HStack } from '@chakra-ui/react'
+import MandatorySymbol from 'app/layouts/components/form/mandatory-symbol'
+import { useEffect, useState } from 'react'
+import { useIsFieldNotEmpty } from 'utils/contexts/IsFieldNotEmptyContext'
+import { useIsProductInternmatch } from 'utils/helpers/check-product-name'
+import useGetProductName from 'utils/helpers/get-product-name'
+import { isNullOrUndefinedOrEmpty } from 'utils/helpers/is-null-or-undefined'
+import useProductColors from 'utils/productColors'
+
+const ApplicationDetailViewFlag = ({
+  questionCode,
+  data,
+  onSendAnswer,
+  placeholderName: label,
+  mandatory,
+}) => {
+  const realm = useGetProductName().toLowerCase()
+  const isProductInternmatch = useIsProductInternmatch()
+
+  const [selected, setSelected] = useState(data?.value || false)
+
+  const { labelTextColor } = useProductColors()
+
+  const { dispatchFieldMessage } = useIsFieldNotEmpty()
+
+  const handleToggle = () => {
+    onSendAnswer(!selected)
```

Comment:
The component should send the labels as the value, in our scenario, the answer that needs to be sent back should be 'approved' and 'rejected' provided they are the labels of the toggle field. If the requirement is to send a boolean value instead for this case, write a helper function that checks the label and returns true if approved and false if rejected. Compose the helper function with the selected value and we can send boolean value for specific scenarios, the actual label otherwise by default.

Reply:
I personally disagree, functionally this component is just a true/false, and it can be see as the question asking "Is this approved?" to which the answer is true/false. I personally believe using labels like this would be unnecessary code complexity